src/Common/src/TypeSystem/Interop/MethodDesc.Interop.cs

```diff
@@ -28,6 +28,40 @@ public virtual PInvokeMetadata GetPInvokeMethodMetadata()
         {
             return default(PInvokeMetadata);
         }
+
+        /// <summary>
+        /// If <see cref="IsPInvoke"/> is true, retrieves the parameters of the platform invoke.
+        /// </summary>
+        public virtual PInvokeArgumentMetadata[] GetPInvokeArgumentMetadata()
+        {
+            return default(PInvokeArgumentMetadata[]);
+        }
+    }
+
+    [Flags]
+    public enum PInvokeArgumentAttributes
+    {
+        None = 0,
+        In = 1,
+        Out = 2,
+        Lcid = 4,
+        Retval = 8,
+        Optional = 16,
+        HasDefault = 4096,
+        HasFieldMarshal = 8192
+    }
+
+    public struct PInvokeArgumentMetadata
```

Comment:
This struct also needs an `Index` field - otherwise there's no way to map it to a parameter number (the code you have right now expects that there's an ordinal mapping, but it's not like that - e.g. the return "parameter" can have marshalling info too, and a source to IL compiler might choose not to emit a record for a parameter that is not "interesting").